docs/weave-tutorial/03-codec.md

```diff
@@ -12,108 +12,118 @@ Weave defines `Marshaller` and `Persistent` interface standards. These interface
 
 ```protobuf
-// Trade is a settled partial/full order
-// We store these as independent entities to help with queries to map
-// the prices over time. They are also referenced by the Orders, so we can
-// see how much was fulfilled.
-//
-// Ask and Bid tickers are "inherited" from the order book
-message Trade {
+message Blog {
   weave.Metadata metadata = 1;
+  // ID is blog's identifier
   bytes id = 2 [(gogoproto.customname) = "ID"];
-  bytes order_book_id = 3 [(gogoproto.customname) = "OrderBookID"];
-  bytes order_id = 4 [(gogoproto.customname) = "OrderID"];
-  // Address of taker (this is an order that was instantly fulfilled)
-  bytes taker = 5 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
-  // Address of maker (this is an order that was stored first before fulfillment)
-  bytes maker = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
-  // this is how much each side paid (they got the opposite one)
-  coin.Coin maker_paid = 7;
-  coin.Coin taker_paid = 8;
-  // executed_at defines execution time of an order
-  int64 executed_at = 9 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
+  // Owner is the owner address if the blog
+  bytes owner = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+  // Title is title of the blog
+  string title = 4;
+  // Description is description section of the blog
+  string description = 5;
+  // CreatedAt defines creation time of the blog
+  int64 created_at = 6 [(gogoproto.casttype) = "github.com/iov-one/weave.UnixTime"];
 }
 ```
 
-As you can see above, weave is heavily using `bytes` for identities, addresses, etc. At first glance, this might seem difficult to handle but if you take a look at [x/orderbook/bucket](https://github.com/iov-one/tutorial/blob/master/x/orderbook/bucket.go#L125) you can see it is useful for indexing and performance.
+<!---
+TODO refactor here when bucket is merged
+-->
+
+As you can see above, weave is heavily using `bytes` for identities, addresses, etc. At first glance, this might seem difficult to handle but if you take a look at [x/blog/bucket](https://github.com/iov-one/tutorial/blob/master/x/orderbook/bucket.go#L125) you can see it is useful for indexing and performance.
 
 Note that the package defined in the protobuf file must match the package name used by the Go language code in the same directory.
 
 You can also import types from one proto file into another. Make sure to use the full github path so that the generated Go code has properly set imports. The package name above is also used as a namespace for the imported protobuf definitions. This is how [x/cash](https://github.com/iov-one/weave/blob/master/x/cash/codec.proto) creates a wallet that contains an array of tokens of different currencies.
```

Comment:
```suggestion
You can also import types from one proto file into another. Make sure to use the full Github path so that the generated Go code has properly set imports. The package name above is also used as a namespace for the imported protobuf definitions. This is how [x/cash](https://github.com/iov-one/weave/blob/master/x/cash/codec.proto) creates a wallet that contains an array of tokens of different currencies.
```